stereotype/contrib/schematics.py

```diff
@@ -68,3 +104,33 @@ def _iterate_validation_errors(messages: dict) -> Iterable[PathErrorType]:
         else:
             for path, error in _iterate_validation_errors(container):
                 yield (str(key),) + path, error
+
+
+def raise_conversion_error(error):
```

Comment:
I found this function quite difficult to understand, so I took the liberty of refactoring it, please see my commit's description for more info.